### PR TITLE
home: don't show blocked followers

### PR DESCRIPTION
**Pull request by maxtaco**
- read the new gregor item user.blocked
- if the blocked UIDs appear in our home state in any place, then flush the state and refetch it
- add a test
- needs https://github.com/keybase/keybase/pull/4749 to pass CI

### CODE REVIEW

**Comment by maxtaco** [comment]
Now needs https://github.com/keybase/keybase/pull/4756 to pass

**Comment by maxtaco** [comment]
@jzila maybe you want to take this while @zapu is AFK, and get it in sooner rather than later

**Comment by jzila** [comment]
Yes sir, will review in an hour or so.

File: go/home/home.go [hunk 1]
```diff
@@ -523,3 +523,49 @@ func (h *Home) pollOnce(m libkb.MetaContext) (d time.Duration, err error) {
...
+		m.Debug("nothing to do, people cache is empty")
+		return false
+	}
+
+	if findBadUserInUsers(m, p.all, badUIDs) {
+		m.Debug("Found blocked user in people cache (all)")
+		return true
+	}
+
+	if findBadUserInUsers(m, p.lastShown, badUIDs) {
```
Comment (jzila) [inline go/home/home.go]: Won't `p.lastShown` be covered by `p.all`?

File: go/service/user_handler.go [hunk 2]
```diff
@@ -77,6 +88,30 @@ func (r *userHandler) passphraseStateUpdate(m libkb.MetaContext, cli gregor1.Inc
...
+	var msg keybase1.UserBlockedGregorBody
+	if err := json.Unmarshal(item.Body().Bytes(), &msg); err != nil {
+		m.Warning("error unmarshaling user.blocked item: %s", err)
+		return err
+	}
+	m.Debug("Got user.blocked item: %+v", msg)
+	badUIDs := make(map[keybase1.UID]bool)
+	for _, r := range msg.Blocks {
+		if (r.Chat != nil && *r.Chat) || (r.Follow != nil && *r.Follow) {
+			badUIDs[r.Uid] = true
```
Comment (jzila) [inline go/service/user_handler.go]: Just want to make sure we've thought about the implications of treating users as a "bad user" that have `r.Follow` but not `r.Chat` blocks set. It seems that in the context of this PR it's all working properly (the user doesn't appear in recommendations). But the future should be considered.

File: go/systests/home_test.go [hunk 3]
```diff
@@ -87,18 +87,25 @@ func findFollowerInHome(t *testing.T, home keybase1.HomeScreen, f string) (prese
...
+		}
+		switch ptyp {
+		case keybase1.HomeScreenPeopleNotificationType_FOLLOWED:
+			follow := people.Followed()
+			if follow.User.Username == f {
+				return true, item.Badged
 			}
-			if typ == keybase1.HomeScreenPeopleNotificationType_FOLLOWED {
-				follow := people.Followed()
+		case keybase1.HomeScreenPeopleNotificationType_FOLLOWED_MULTI:
```
Comment (jzila) [inline go/systests/home_test.go]: Is this a bugfix?

File: go/systests/home_test.go [hunk 4]
```diff
@@ -230,3 +237,43 @@ func attachHomeUI(t *testing.T, g *libkb.GlobalContext, hui keybase1.HomeUIInter
...
+			var present bool
+			present, _ = findFollowerInHome(t, home, u.username)
+			return (present == shouldBePresent)
+		})
+	}
+
+	probeUserInAliceHome(bob, true)
+	probeUserInAliceHome(charlie, true)
+
+	alice.block(bob.username, true /*chat*/, true /*follow*/)
```
Comment (jzila) [inline go/systests/home_test.go]: Would be nice to get a test with just "follow" and just "chat" blocks set.

File: go/home/home.go (hunk 1 above)
Comment (maxtaco) [reply go/home/home.go]: I think so, but I wasn't sure.  Figure it didn't hurt to take the conservative approach.

File: go/service/user_handler.go (hunk 2 above)
Comment (maxtaco) [reply go/service/user_handler.go]: I just figured if you clicked block on the chat, you didn't want to then see them on your home screen.

File: go/systests/home_test.go (hunk 3 above)
Comment (maxtaco) [reply go/systests/home_test.go]: It only breaks for tests that have two follows in a row, and this PR has the first one.

File: go/home/home.go (hunk 1 above)
Comment (jzila) [reply go/home/home.go]: Understood, please include a comment mentioning that, since others perusing this code might be confused like I was.

**Comment by maxtaco** [comment]
thanks @jzila

**Comment by maxtaco** [comment]
Will merge this one and the rebase the next one on top of it, after merge. I'm not counting an a CI green though since there seems to be a chat flake

**Pull request merged**